sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala

```diff
@@ -439,6 +439,13 @@ case class MapObjects(
       s"boolean ${loopVar.isNull} = ${genInputData.isNull} || ${loopVar.value} == null;"
     }
 
+    // If lambdaFunction is WrapOption, we will not determine null or not based on the
+    // value of loopVar.isNull, because WrapOption will return None for null.
+    val isWrapOption = lambdaFunction match {
```

Comment:
Let me explain it.

When we pass in an array with None. It will be encoded as null internally. When we decode it back,  WrapOption is called to re-construct it.

The logic of MapObjects is to assign an element as null if its given input element is null. So It will not actually go into WrapOption to re-construct a None back. In order to do that, we need to call lambdaFunction even the element is null.

But we can't simply ignore loopVar.isNull and call all kinds of lambdaFunctions. I tried before but for some lambdaFunctions, a null input value causes problematic results.

In the end I can only check if lambdaFunction is WrapOption or not to make the decision here. Do you have other suggestion other than a hack like this here? Thanks.